### PR TITLE
feat: rename to aspera SDK

### PR DESCRIPTION
**Pull request by dabrad26**

### CODE REVIEW

File: docs/DEVELOPMENT.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
-# IBM Aspera Browser TypeScript SDK
+# IBM Aspera TypeScript SDK
```
Comment (gegles) [inline docs/DEVELOPMENT.md:1]: Should it be TypeScript or JavaScript here?

File: docs/DEVELOPMENT.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
 
-Enables web applications to utilize Aspera file-transfer capabilities using IBM Aspera via the browser.
+Enables web applications to utilize Aspera file-transfer capabilities using IBM Aspera via JavaScript.
```
Comment (gegles) [inline docs/DEVELOPMENT.md:3]:
```suggestion
This JavaScript SDK enables web applications to utilize Aspera file-transfer capabilities via the IBM Aspera App for Desktop.
```

**Review by dwosk — APPROVED** [review]

File: src/helpers/ws.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
 import {messages} from '../constants/messages';
-import {asperaBrowser} from '../index';
-import {TransferResponse} from '../models/aspera-browser.model';
+import {asperaSdk} from '../index';
+import {TransferResponse} from '../models/aspera-sdk.model';
 import {WebsocketEvent, WebsocketMessage, WebsocketTopics} from '../models/models';
 
 export class WebsocketService {
-  /** The main websocket connection to Aspera Browser */
+  /** The main websocket connection to Aspera */
```
Comment (gegles) [inline src/helpers/ws.ts:8]:
```suggestion
  /** The main websocket connection to Aspera App*/
```

**Review by gegles — CHANGES_REQUESTED** [review]

File: src/helpers/client/http-client.ts [hunk 4]
```diff
@@ -64,7 +69,7 @@ class JSONRPCHttpClient {
 }
 
 /**
- * Client used for making requests to Aspera Browser.
+ * Client used for making requests to Aspera.
```
Comment (gegles) [inline src/helpers/client/http-client.ts:72]:
```suggestion
 * Client used for making requests to IBM Aspera App.
```

File: src/models/aspera-sdk.model.ts [hunk 5]
```diff
@@ -321,20 +321,20 @@ export class Browser {
...
   /** Function to display a folder dialog for the user to select folders. */
   showSelectFolderDialog: (options?: FolderDialogOptions) => Promise<DataTransferResponse>;
-  /** Function to display the IBM Aspera Browser preferences page */
+  /** Function to display the IBM Aspera preferences page */
   showPreferences: () => Promise<any>;
   /** Function to modify a running transfer */
-  modifyTransfer: (transferId: string, options: ModifyTransferOptions) => Promise<BrowserTransfer>;
-  /** Function to set custom branding for IBM Aspera Browser */
+  modifyTransfer: (transferId: string, options: ModifyTransferOptions) => Promise<AsperaSdkTransfer>;
+  /** Function to set custom branding for IBM Aspera */
```
Comment (gegles) [inline src/models/aspera-sdk.model.ts:337]:
```suggestion
  /** Function to set custom branding for IBM Aspera App*/
```

File: src/app/installer.ts [hunk 6]
```diff
@@ -1,18 +1,18 @@
-import {asperaBrowser} from '../index';
+import {asperaSdk} from '../index';
 import {errorLog, generateErrorBody, generatePromiseObjects, getCurrentPlatform, isValidURL, throwError} from '../helpers/helpers';
 import {apiGet} from '../helpers/http';
 import {messages} from '../constants/messages';
 import {InstallerInfoResponse, InstallerOptions} from '../models/models';
 
 /**
- * Get the latest Aspera Browser installer information such as download URL and version.
+ * Get the latest Aspera installer information such as download URL and version.
```
[the remark this answers is not in the document]
Comment (dwosk) [reply src/app/installer.ts:8]: This is for the desktop app installer - don't think it needs to say "Aspera SDK installer"?

File: tests/installer.spec.ts [hunk 7]
```diff
@@ -55,7 +55,7 @@ describe('getInstallerInfo', () => {
         'platform': 'macos',
         'type': 'dmg',
         'arch': 'universal',
-        'url': 'https://d3gcli72yxqn2z.cloudfront.net/downloads/desktop/macos/1.2.0/stable/universal/ibm-aspera-browser_1.2.0_macos.dmg'
+        'url': 'https://d3gcli72yxqn2z.cloudfront.net/downloads/desktop/macos/1.2.0/stable/universal/ibm-aspera-sdk_1.2.0_macos.dmg'
```
Comment (gegles) [inline tests/installer.spec.ts:58]: Those links are wrong... probably some bad search and replace

**Comment by dabrad26** [comment]
We can always improve the incode docs (TBH the docs shouldn't have so much use of `Aspera this and that` since in the context of coding that it clear.  

But I need to get a build to AOC otherwise we will end up having to wait until next prod push and not be able to get the new Safari stuff in app.

File: tests/installer.spec.ts [hunk 8]
```diff
@@ -85,35 +85,35 @@ describe('getInstallerInfo', () => {
...
+          'url': 'downloads/ibm-aspera-sdk_1.1.9.AppImage'
         }
       ]
     };
     (<any>global).fetch = mockFetch(response);
-    const data = await getInstallerInfo({endpoint: 'https://aspera.us/aspera/browser', all: true});
+    const data = await getInstallerInfo({endpoint: 'https://aspera.us/aspera/sdk', all: true});
     for (const entry of data.entries) {
-      expect(entry.url.startsWith('https://aspera.us/aspera/browser/downloads/ibm-aspera-browser')).toBe(true);
+      expect(entry.url.startsWith('https://aspera.us/aspera/sdk/downloads/ibm-aspera-sdk')).toBe(true);
```
Comment (gegles) [inline tests/installer.spec.ts:116]: Wrong links?

File: tests/installer.spec.ts [hunk 9]
```diff
@@ -125,7 +125,7 @@ describe('getInstallerInfo', () => {
           'platform': 'macos',
           'type': 'dmg',
           'arch': 'universal',
-          'url': 'downloads/ibm-aspera-browser_1.2.0_macos.dmg'
+          'url': 'downloads/ibm-aspera-sdk_1.2.0_macos.dmg'
```
Comment (gegles) [inline tests/installer.spec.ts:128]: Wrong link

File: typedoc.js [hunk 10]
```diff
@@ -12,6 +12,6 @@ module.exports = {
   suppressExcessPropertyErrors: 'true',
   suppressImplicitAnyIndexErrors: 'true',
   module: 'commonjs',
-  name: 'IBM Aspera Browser JavaScript Library',
+  name: 'IBM Aspera JavaScript Library',
```
Comment (gegles) [inline typedoc.js:15]:
```suggestion
  name: 'IBM Aspera JavaScript SDK',
```
Maybe?

File: webpack.config.js [hunk 11]
```diff
@@ -40,11 +40,11 @@ module.exports = {
         directory: './src',
       },
       {
-        publicPath: '/aspera-browser-sdk-js/docs',
+        publicPath: '/aspera-sdk-js/docs',
         directory: './dist/js/docs',
       },
     ],
-    host: 'browser-sdk.aspera.us',
+    host: 'js-sdk.aspera.us',
```
Comment (gegles) [inline webpack.config.js:47]:
```suggestion
    host: 'sdk.aspera.us',
```
Not quite what this is...

**Review by gegles — CHANGES_REQUESTED** [review]

File: tests/installer.spec.ts (hunk 7 above)
Comment (dabrad26) [reply tests/installer.spec.ts:58]: No. They are just fake links for unit tests.

File: webpack.config.js (hunk 11 above)
Comment (dabrad26) [reply webpack.config.js:47]: This is for internal dev running. Just needs to be a little unique as we use this for other JavaScript projects like http gateway and aoc stuff.

File: tests/installer.spec.ts (hunk 9 above)
Comment (dabrad26) [reply tests/installer.spec.ts:128]: Just fake data for unit tests mock

**Pull request merged**